game/games/sortit/player.gd

```diff
@@ -0,0 +1,184 @@
+class_name SortItPlayer
+extends KinematicBody
+
+export(bool) var old_rotational_controls = false
+export(float) var direction_change_speed = 4
+export(int) var move_speed = 1000
+export(float) var max_speed = 5
+export(float) var atrract_speed = 500
+export(float) var friction = 0.7
+export(float) var angular_friction = 0.5
+export(float) var max_box_distance = 5.0
+export(float) var attracted_distance = 2.7
+export(float) var camera_rotate_speed = 0.8
+export(Vector3) var camera_offset = Vector3(-8, 20, 0)
+export(Material) var greater_compare_material
+export(Material) var normal_compare_material
+
+# Needs to be set by parrent class
+var player_index: int
+var camera: Camera
+var status_display: Control
+
+var left_magnet_active = false
+var right_magnet_active = false
+
+var left_box = null
+var right_box = null
+
+var _last_direction = Vector3.ZERO
+var _velocity = Vector3.ZERO
+var _angular_velocity = 0.0
+onready var _left_anchor = $LeftMagnet/Anchor
+onready var _right_anchor = $RightMagnet/Anchor
+onready var _players = get_parent()
+onready var _main_body_mesh = $Mesh/MainBody
+
+
+func set_color(color: Color):
+	var material: SpatialMaterial = _main_body_mesh.mesh.surface_get_material(0).duplicate()
+	material.albedo_color = color
+	_main_body_mesh.mesh.surface_set_material(0, material)
+
+
+func _input(_event):
```

Comment:
The bug:
- Its not possible to to drop both cubes at the same time.
1. _players.is_action_just_pressed is somehow called on release
2. q and e are somehow connected (try hold q and press e)
3. no InputEventKey echo handling

I tried to fix it, but I gave up after an hour.

Reply:
Some of this is important, but the chance, that a player presses two buttons in the exact same frame is quite low.

Reply:
You miss the point. Holding the button shows that there is a connection between q and e.
And it don't have to be in the same frame.

Reply:
Ok, maybe I phrased my response a bit confusingly. My argument is: I never once got this bug while actually playing the game (Not looking for bugs). Because I never pressed the buttons in that order and time while playing. Secondly, I have already tried to fix this bug, but I got stuck too. I do agree that this is not optimal, but currently I don't have the capacity to fix a bug, that does not occur that often. I will probably try to fix this again, but I will probably do this in a second PR later.

Reply:
found it https://github.com/RedstoneMedia/Suffragium/pull/1

Reply:
?